test/jdk/java/lang/reflect/code/TestPrimitiveTypePatterns.java

```diff
@@ -0,0 +1,120 @@
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Method;
+import java.lang.reflect.code.*;
+import java.lang.reflect.code.bytecode.BytecodeGenerator;
+import java.lang.reflect.code.interpreter.Interpreter;
+import java.lang.reflect.code.op.CoreOp;
+import java.lang.reflect.code.type.JavaType;
+import java.lang.reflect.code.type.PrimitiveType;
+import java.lang.runtime.CodeReflection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.lang.reflect.code.op.CoreOp.*;
+import static java.lang.reflect.code.op.ExtendedOp.*;
+import static java.lang.reflect.code.type.FunctionType.*;
+
+/*
+ * @test
+ * @run testng TestPrimitiveTypePatterns
+ * @enablePreview
+ */
+
+public class TestPrimitiveTypePatterns {
+
+    @DataProvider
+    public static Object[][] patternsOfInt() {
+        return new Object[][]{
+                {JavaType.BYTE, new int[] {Byte.MIN_VALUE, Byte.MAX_VALUE, Byte.MIN_VALUE -1, Byte.MAX_VALUE + 1}},
+                {JavaType.SHORT, new int[] {Short.MIN_VALUE, Short.MAX_VALUE, Short.MIN_VALUE -1, Short.MAX_VALUE + 1}},
+                {JavaType.CHAR, new int[] {Character.MIN_VALUE, Character.MAX_VALUE, Character.MIN_VALUE -1, Character.MAX_VALUE + 1}},
+        };
+    }
+
+    @Test(dataProvider = "patternsOfInt")
+    void testPatternsOfInt(JavaType targetType, int[] values) throws Throwable {
+
+        var genericModel = getCodeModel("f");
+
+        var model = buildTypePatternModel(genericModel, JavaType.INT, targetType);
+        model.writeTo(System.out);
+
+        var lmodel = model.transform(OpTransformer.LOWERING_TRANSFORMER);
+        lmodel.writeTo(System.out);
+
+        var mh = BytecodeGenerator.generate(MethodHandles.lookup(), lmodel);
+
+        for (int v : values) {
+            Assert.assertEquals(Interpreter.invoke(lmodel, v), mh.invoke(v));
+        }
+    }
+
+    @CodeReflection
+    // works as generic model that will be transformed to test conversion from a sourceType to targetType
+    static boolean f(byte b) {
```

Comment:
Alas unfortunately such a transformation it is harder than i anticipated because the types permeate through pattern match operation, so you have to replace the whole pattern match op. Which means you might as well build the model explicitly, sorry if i misled you.

Reply:
we are only replacing the patter type op

Reply:
I see what you mean.